x/incentives/keeper/gauge.go

```diff
@@ -221,6 +221,13 @@ func (k Keeper) GetLocksToDistribution(ctx sdk.Context, distrTo lockuptypes.Quer
 	return []lockuptypes.PeriodLock{}
 }
 
+// GetLocksToF1Distribution Get locks that are eligible to get gauge incentives
+func (k Keeper) GetLocksToF1Distribution(ctx sdk.Context, denom string, duration time.Duration) []lockuptypes.PeriodLock {
+	epochInfo := k.GetEpochInfo(ctx)
+	timestamp := epochInfo.CurrentEpochStartTime.Add(epochInfo.Duration).Add(duration)
```

Comment:
Why is the duration added two times?

Reply:
GetLocksToF1Distribution() is a function to get locks which are eligible to get gauge rewards, which means unlocking locks should have remaining time longer than gauge.duration
when epochInfo retrieved by GetEpochInfo(ctx) contains data of last epoch
(https://github.com/osmosis-labs/osmosis/blob/0a105c9608a23f2f333811e1557da04da5ce8d03/x/epochs/abci.go#L14)
so unlocking lock.endTime should be longer than current EpochTime + gauge.duration == last epoch time+epoch.duration + gauge.duration